python/pyarrow/array.pxi

```diff
@@ -1009,7 +1009,8 @@ cdef class Array(_PandasConvertible):
             return NotImplemented
 
     def equals(Array self, Array other):
```

Comment:
Why not:
```python
    def equals(Array self, Array other, *, nans_equal=True):
```

Reply:
The ship might have sailed on this (I don't know how long this is already exposed in the public C++ API), but numpy has this as `equal_nan` keyword